Inferences/ForwardDemodulation.cpp

```diff
@@ -167,9 +163,23 @@ bool ForwardDemodulationImpl<combinatorySupSupport>::perform(Clause* cl, Clause*
           }
         }
   #endif
-        if(!preordered && (_preorderedOnly || ordering.compare(trm,rhsS)!=Ordering::GREATER) ) {
+
+        auto subs = qr.unifier;
+        ASS(subs->isIdentityOnQueryWhenResultBound());
+
+        Applicator<true> varSubst(subs.ptr(), subst);
+        Applicator<false> notVarSubst(subs.ptr(), subst);
```

Comment:
This is quite a general remark I'd love to see in the general vampire coding style: Could we add comments for boolean arguments when their semantics are not clear from the context?
In this case for example have 
```
        Applicator</* applyVsubst */ true> varSubst(subs.ptr(), subst);
        Applicator</* applyVsubst */ false> notVarSubst(subs.ptr(), subst);
```
Which makes the code more readable in many cases.

Reply:
I added untemplated classes, see new commit.